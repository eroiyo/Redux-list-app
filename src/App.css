* {
  color: white;
  font-family: Gill Sans, Gill Sans MT, Calibri, sans-serif;
}

body {
  margin: 0;
}

.App {
  display: flex;
  background-color: #568ce4;
  width: 100vw;
  height: 100vh;
}

.container {
  display: flex;
  flex-direction: column;
  width: 100vw;
  height: 100vh;
}

.filtered {
  filter: invert(12%) sepia(45%) saturate(1041%) hue-rotate(181deg)
    brightness(92%) contrast(100%);
}
