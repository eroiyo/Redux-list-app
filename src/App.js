import React, { useState } from 'react';
import {
  Redirect, Route, BrowserRouter, Switch,
} from 'react-router-dom';
import './App.css';
import Categorys from './components/categorys/Categorys';
import Navbar from './components/Navbar/Navbar';
import Displayer from './components/Displayer/Displayer';
import Separator from './components/Separator/Separator';
import List from './components/List/List';

const App = () => {
  const [currentPosition, setData] = useState(undefined);

  return (
    <div className="App">
      <div className="container">
        <BrowserRouter>
          <Navbar setData={setData} />
          <Displayer show={currentPosition} />
          <Separator show={currentPosition} />
          <Switch>
            <Route exact path="/">
              <Categorys setData={setData} />
            </Route>
            <Route path="/product">
              <List show={currentPosition} />
            </Route>
            <Redirect to="/" />
          </Switch>
        </BrowserRouter>

      </div>
    </div>
  );
};
export default App;
