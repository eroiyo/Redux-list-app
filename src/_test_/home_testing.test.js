import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import App from '../App';

const ReactTestRenderer = require('react-test-renderer');

it('Take a snaphot, and compared it, to see if they are equal', () => {
  const home = ReactTestRenderer.create(<Provider store={store}><App /></Provider>).toJSON();
  expect(home).toMatchSnapshot();
});
