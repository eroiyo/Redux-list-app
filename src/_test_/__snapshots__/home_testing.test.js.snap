// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take a snaphot, and compared it, to see if they are equal 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <nav
      className="navBar"
    >
      <div
        className="header"
      >
        <a
          active="true"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <img
            alt="return"
            className="arrow-left"
            src="next.svg"
          />
        </a>
        <h6
          className="header-title"
        >
          FakeShop
        </h6>
        <img
          alt="mic"
          className="mic"
          src="microphone.svg"
        />
        <img
          alt="settings"
          className="settings"
          src="settings.svg"
        />
      </div>
    </nav>
    <div
      className="Displayer"
    >
      <div
        className="display-b img filtered"
      >
        <img
          alt="displayer"
          src="s.svg"
        />
      </div>
      <div
        className="display-b text"
      >
        <h1>
          FakeShop
        </h1>
        <p>
          0
           
          Results
        </p>
      </div>
    </div>
    <div
      className="Separator"
    >
      <div
        className="inside"
      >
        <h6
          className="header-title separator-title"
        >
          Categorys
        </h6>
      </div>
    </div>
    <div
      className="Categorys"
    >
       
    </div>
  </div>
</div>
`;
